components/org.wso2.carbon.identity.scim2.provider/src/main/java/org/wso2/carbon/identity/scim2/provider/resources/GroupResource.java

```diff
@@ -333,29 +462,144 @@ private Response processRequest(final Map<String, String> requestAttributes) {
                 // Processing count and startIndex in the request.
                 Integer startIndex = convertStringPaginationParamsToInteger(
                         requestAttributes.get(SCIMProviderConstants.START_INDEX), SCIMProviderConstants.START_INDEX);
-                Integer count = convertStringPaginationParamsToInteger(requestAttributes.get(SCIMProviderConstants.COUNT),
-                        SCIMProviderConstants.COUNT);
+                Integer count = convertStringPaginationParamsToInteger(requestAttributes.get(SCIMProviderConstants
+                        .COUNT), SCIMProviderConstants.COUNT);
                 scimResponse = groupResourceManager
                         .listWithGET(userManager, filter, startIndex, count, sortBy, sortOrder, domainName, attributes,
                                 excludedAttributes);
+            } else if (GET.class.getSimpleName().equals(httpVerb) && requestAttributes.get(SCIMProviderConstants
+                    .ATTRIBUTES).equals(PERMISSIONS)) {
```

Comment:
Shall we extract this to a private method for readability. Ex. isGroupPermissionsGetRequest()